assets/app/actions/siteActions.js

```diff
@@ -55,6 +55,23 @@ export default {
     }).catch(error => alertActions.httpError(error.message));
   },
 
+  uploadFile(site, path, fileData, message) {
```

Comment:
This action creator is very similar to the createCommit action below it. I'd like to be able to unify them, but I'm not sure of the best way to do that.

They diverge in two major ways
* the file data is already base64 encoded when uploading an asset
* we have different callback behavior in each of the two actions &emdash; the uploadFile action doesn't redirect after a successful upload, but the createCommit action does